examples/seismic/acoustic/gradient_example.py

```diff
@@ -1,109 +1,147 @@
 import numpy as np
 from numpy import linalg
+from cached_property import cached_property
 
 from devito import TimeFunction, Function
 from examples.seismic import Receiver, RickerSource, demo_model
 from examples.seismic.acoustic import ForwardOperator, GradientOperator, smooth10
 
+class GradientExample(object):
+    def __init__(self, shape=(50, 50, 50), spacing=(15.0, 15.0, 15.0), tn=500., time_order=2,
+            space_order=4, nbpml=10):
+        self.time_order = time_order
+        self.space_order = space_order
+        self._setup_model_and_acquisition(shape, spacing, nbpml, tn)
+        self._true_data()
```

Comment:
what's going here?

Reply:
Assuming your question is about the method call `self._true_data()`, in a realistic workflow this method would load the reflection data recorded in the real experiment from disk. Since this is artificial, we run a forward simulation inside this method and store the simulated data on the object.